### PR TITLE
Prep for releasing jars

### PR DESCRIPTION
**Pull request by big-andy-coates**
Tidy up dependencies a bit, fix a few issues and prep for publishing jars

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -132,14 +127,10 @@ subprojects {
         }
     }
 
-    tasks.register("pre") {
+    tasks.register("format") {
```
Comment (big-andy-coates) [inline build.gradle.kts:130]: renamed to match Creek - otherwise it'll drive me bonkers!  Hope that's OK.

File: build.gradle.kts [hunk 2]
```diff
@@ -238,4 +238,6 @@ subprojects {
 
         sign(publishing.publications["mavenArtifacts"])
     }
-}
\ No newline at end of file
+}
+
+defaultTasks("format", "static", "check")
```
Comment (big-andy-coates) [inline build.gradle.kts:243]: This means you can just run `./gradlew` from the terminal to format and check the code.

(Still do a `./gradlew build` before committing as that'll also check javadoc etc.)

File: build.gradle.kts (hunk 1 above)
Comment (bluemonk3y) [reply build.gradle.kts:130]: no worries

File: build.gradle.kts (hunk 2 above)
Comment (bluemonk3y) [reply build.gradle.kts:243]: no worries

**Review by bluemonk3y — APPROVED** [review]
lgtm

**Pull request merged**